### DISCUSSION TITLE
Should we be concerned about the apparent lack of development or urgency to fix issues?

### DISCUSSION THREAD

**Discussion by chrisdeeming** [postalserver/postal#2744] [Help with using Postal] [answered]
I am completely new here so I apologise in advance for what might appear to be a somewhat antagonistic first contact. I'm excited to get Postal configured for use in our production environments and on the surface I think this is a fantastic piece of work that shows a lot of promise.

But as I dive a little deeper, I have a few things starting to cause me a little bit of hesitation.

At the time of writing this, the last release was over 8 months ago.

There have been some bug fixes which were fixed in that time, but for reasons I don't quite understand, they haven't been pushed out into a new stable release.

For example, #2573 feels like a fairly significant issue. This seems to have broken some relied upon functionality which would have prevented existing set ups from sending emails in some cases. If we were to have faced a bug like this in our production environments it would be affecting millions of our users. It was quickly identified and fixed but that fix is not yet available in the current release. This was fixed over 7 months ago.

Thankfully there is a workaround so I am not immediately concerned.

Incidentally, is there a way to update Postal server to pull from the current development branch rather than the last tagged version?

I think this project could benefit from some frank and candid communication from the maintainers so that its user base is kept assured of its future. I wholeheartedly hope that its future is a bright one as it very much deserves.

**Comment by PrabhuShanmughapriyan** [comment]
@chrisdeeming - For the prod release , you should always tagged versions as mentioned here - https://docs.postalserver.io/other/containers

**Comment by Pacerino** [comment]
As someone who also installed Postal on a production environment about a month ago, I can fully understand your concerns and incentive. We send around 1.6 million emails a month and it would be fatal if the mail server stopped working from one day to the next. However, the problem exists, in theory, with any software.

For me, however, the current bugs that exist are not a problem. Sure, the strange formatting of the mails in the overview is a bit annoying, but it doesn't interfere with sending the mails.

For error [2153](https://github.com/postalserver/postal/issues/2153), for example, I simply implemented the change and rebuilt the Docker container in our repository.

To answer your last question, that's exactly what I would do: I would simply build the branch locally on my machine using Docker. The `ci.yml` from Github provides me with the necessary steps for this.

I hope I was able to give you an insight into how I deal with the situation of this project. Like you, I hope that this project has a great future.

**Comment by chrisdeeming** [comment]
Thanks for the replies.

We've gone all in and we're now using Postal in production. A few hurdles that weren't insurmountable, but on the whole I am happy.

I want nothing more than for this project to succeed, more so than ever now, but I am still concerned and I think to an extent we all should be.

There are some very early warning signs that cannot be ignored. @adamcooke and other maintainers no longer seem to be active here. I even emailed Krystal directly for comment and this email has gone unanswered.

This is a great project with so much potential and it would be a shame to see that wasted. This kind of application is at the level of utility and overall quality of a commercial application. I'm sure the majority of people using Postal would be happy to secure its future by financially backing it if that were even an option. But it isn't an option and I worry that may mean the incentive isn't there for the developers to continue its development or take it seriously.

But I'm getting ahead of myself. Future development would be great but what has caused this uncertainty is very simple to fix:

* Periodic releases for fixed issues
* Occasional support for issues from maintainers (as was traditionally the case)
* Pull requests being reviewed

These three things would go a long way to increase confidence and would set my mind at rest entirely and surely should not be considered an unreasonable expectation.

**Comment by chrisdeeming** [comment, marked as answer]
Just to close this off, the recent activity is a welcome sight. On the whole, despite being a bit grumpy (out of uncertainty), I am happy I went all-in on Postal. Whatever happens it is a great tool and I'm sure it has a long and bright future. I wish everyone involved well and can't thank them enough for the effort that has gone into this.
